04-using-github-with-the-platform.Rmd

```diff
@@ -220,35 +224,41 @@ A Personal Access Token (PAT) is provided by GitHub through each user's profile,
+Conveniently, most [`r-lib`-style](https://github.com/r-lib) packages (those built by Hadley Wickham and friends--for most people a backbone of R infrastructure) including `remotes` and `devtools`, will look for a PAT in the system environment by default, under the name `GITHUB_PAT`, so no modifications need be made to any project code once a user has registered a PAT in this location.
 This is the approach we'll make use of:
 
   1. Set the system environment variable `GITHUB_PAT` by pasting in your PAT Token into the below:
-  ```{r, eval = FALSE}
-  Sys.setenv("GITHUB_PAT" = "ax451...8838b1")
-  ```
   
+      ```{r, eval = FALSE}
+      Sys.setenv("GITHUB_PAT" = "ax451...8838b1")
+      ```
+
   2. Any call to `remotes::install_github` will use it without having to be asked!
-  ```{r, eval = FALSE}
-  remotes::install_github("moj-analytical-services/[private-package]@v1.6")
-  ```
+
+      ```{r, eval = FALSE}
+      remotes::install_github("moj-analytical-services/[private-package]@v1.6")
+      ```
+
 This approach means that install lines like (2) above can be freely saved in shared code without compromising GitHub security.
 
-### Advanced 
-The way `remotes::install_github` is through the parameter `auth_token`, to which the PAT can be supplied manually as a string if desired. The default is a call to `auth_token = github_pat()` which simply searches for a PAT in the system environment as above.
-However, this approach leaves users **liable to leaving the PAT lying around in code**. As discussed above this is **not a good idea,** not to mention the fact that it requires the PAT to be copied-and-pasted a number of times across different projects and analyses. 
+**Security reminder**
+
+Never put a secret or password in your code. Even when the repo is private. See MOJ policy: https://ministryofjustice.github.io/security-guidance/standards/secrets-management/#application--infrastructure-secrets
 
 ### A Note on Impermanence
-Because of the pod system on which the Platform works, system environment variables are _not_ always as persisent as hoped. It's worth putting the `setenv` call above (1) into an `.RProfile` or other file **that you do not upload to GitHub** to give easy, but private access. Keep a copy of your PAT on your local machine. You can use `Sys.getenv("GITHUB_PAT")` to check whether a PAT is loaded.
 
-## Other tips and tricks [Work in progress!]:
+Because of the pod system on which the Platform works, system environment variables are _not_ always as persistent as hoped. It's worth putting the `setenv` call above (1) into an `.RProfile` or other file **that you do not upload to GitHub** to give easy, but private access. Keep a copy of your PAT on your local machine. You can use `Sys.getenv("GITHUB_PAT")` to check whether a PAT is loaded.
```

Comment:
Hang on - didn't we just tell them _not_ to keep a copy of the PAT on their local machine?

Also do we really want them to put their `setenv` call into `.Rprofile`? This seems risky because it could easily be accidentally pulled into github. Might it be more secure to ask them to create an `Renv` file or something similar that they can put their `setenv` files in, and then immediately add this to their `.gitignore` to prevent a visit from Mr Cockup?

Reply:
This is now addressed.